### ISSUE TITLE
Frost Mage Winter's Chill issue

### ISSUE THREAD

**Issue by Lipstopica** [Hekili/hekili#471]
I've noticed on Mage after using Flurry it will suggest to use Ice Lance which is correct because of the Winter's Chill target debuff which is 2 stacks so you would double Ice Lance but for some reason it suggest Frostbolt after the first Ice lance with 1 stack left of Winter's Chill. It does change to Ice Lance eventually. But it stays on Frostbolt for some time before switching.
Can be hard to follow when higher haste for example lusting etc.

https://pastebin.com/wzK4zM1S

I hope I captured it correctly. But to reproduce this just follow the rotation it shows and after flurry it will suggest ice lance then into frost bolt then switches to Ice Lance near the end of the gcd.

**Comment by Lipstopica** [comment]
I'm guessing it's kind of meant to be like that because it's predicting at that time you can cast a Frostbolt and then Ice Lance right after, Because I am pretty sure the Ice Lance will reach the target before Frostbolt will, from further testing.

**Comment by Hekili** [comment]
I'll do some testing again.  The addon does model having an Ice Lance in the air (consuming Winter's Chill).  But if Winter's Chill is consumed on Ice Lance cast, then I should undo that.

**Comment by Lipstopica** [comment]
It's one of those where I had to record the gameplay and did see the Ice Lance hit the target first but there was one occasion where it suggest rune of power after the Frostbolt. Will check more myself when I can see if I can get that situation to happen again. But not actually sure if would or or not be a dps increase from how it is currently to if it was moved. I guess with it removed it would be a safer option but not sure at the cost of dps.

The actual debuff consumption is when the Ice Lance hits the target, and as Ice Lance hits the target first before the casted Frostbolt reaches the target (didn't test close distance). It makes sense how it is currently, but I guess it can easily be break the rotation with that situation I had once.
